catalog/async/catalog-async-data-api/src/main/java/org/codice/ddf/catalog/async/data/api/internal/ProcessResource.java

```diff
@@ -61,9 +61,12 @@
   String getMimeType();
 
   /**
-   * Return the input stream containing the {@link ProcessResource}'s actual data content.
+   * Return an input stream containing the {@link ProcessResource}'s actual data content. When this
+   * method is called it creates a {@code TemporaryFileBackedOutputStream} so that it can produce a
+   * new input stream with each call. {@link #close()} should be called once new input streams are
+   * no longer needed.
    *
-   * @return the {@link ProcessResource}'s input stream
+   * @return a copy of the {@link ProcessResource}'s input stream
```

Comment:
✏️ To avoid leaking possible implementation details which you cannot know here since it is an interface, I would limit the documentation to simply something like "a new and independent {@link ProcessResource}'s input stream each time this method is called"